### PR TITLE
fixes #575 Stop merging OR children into intersect nodes

### LINKED ISSUE #575: IndexInfo.intersect() incorrectly includes OR descendants in generated node

### PR DESCRIPTION
**Pull request by FineAndDandy**

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java [hunk 1]
```diff
@@ -468,18 +468,7 @@ protected IndexInfo intersect(long maxPossibilities, Iterable<IndexMatch> matchI
...
-                            topLevelOr = sourceNode;
-                            // add the source node
-                            nodesMap.put(TreeHashVisitor.getNodeHash(sourceNode), node);
-                        } else {
-                            for (int i = 0; i < topLevelOr.jjtGetNumChildren(); i++) {
-                                JexlNode baseNode = getSourceNode(topLevelOr.jjtGetChild(i));
-                                nodesMap.put(TreeHashVisitor.getNodeHash(baseNode), topLevelOr.jjtGetChild(i));
-                            }
-                        }
+                        nodesMap.put(TreeHashVisitor.getNodeHash(node), node);
```
[the remark this answers is not in the document]
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java]: I think that is fine as we are simply removing the reference nodes.  May not be needed if we assume we are flattened, but it does not hurt.

[the remark this answers is not in the document]
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java]: I deleted my previous comment as I read your comment backwards.  Now that I am standing on my feet again.....  I agree with this comment.  Taking out the getSourceNode is perhaps one more change than we really need here.  We need a separate ticket to investigate cleaning this time of code out of the union as well at which time we can evaluate the need for removing reference nodes.

[the remark this answers is not in the document]
Comment (FineAndDandy) [reply warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java]: agree, will put that back in

**Comment by ivakegg** [comment]
This was merged as it was included in pull request #584

**Pull request closed without merging**